xptifw/samples/include/xpti_timers.hpp

```diff
@@ -0,0 +1,95 @@
+//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+#pragma once
+
+#include <atomic>
+#include <sstream>
+#include <thread>
+#include <unordered_map>
+
+namespace xpti {
+class ThreadID {
+public:
+  typedef std::unordered_map<std::string, int> thread_lut_t;
+
+  ThreadID() : m_tid(0) {}
+  ~ThreadID() {}
+
+  inline uint32_t enumID(std::thread::id &curr) {
+    std::stringstream s;
+    s << curr;
+    std::string str(s.str());
+
+    if (m_thread_lookup.count(str)) {
+      return m_thread_lookup[str];
+    } else {
+      uint32_t enumID = m_tid++;
+      m_thread_lookup[str] = enumID;
+      return enumID;
+    }
+  }
+
+  inline uint32_t enumID(const std::string &curr) {
+    if (m_thread_lookup.count(curr)) {
+      return m_thread_lookup[curr];
+    } else {
+      uint32_t enumID = m_tid++;
+      m_thread_lookup[curr] = enumID;
+      return enumID;
+    }
+  }
+
+private:
+  std::atomic<uint32_t> m_tid;
+  thread_lut_t m_thread_lookup;
+};
+
+namespace timer {
+#include <cstdint>
```

Comment:
Curious.
I wonder whether it is allowed by C++ standard to have standard includes in some namespace...